packages/store/tests/state-token.spec.ts

```diff
@@ -15,7 +15,7 @@ describe('[TEST]: StateToken', () => {
   });
 
   describe('Integration', () => {
-    it('should successfully create store', () => {
+    it('should successfully create store', async(() => {
```

Comment:
Does this need to be async btw?

Reply:
@arturovt this spec failed sporadically locally on my laptop.
The spec stopped failing sporadically after I made it `async`.
As far as I know all specs that interact with streams (if we want to assert something after stream emits) should be `async`.

The test component has an observable https://github.com/ngxs/store/blob/461a3346e4ec6d2451eb5a7b181270017a0853db/packages/store/tests/state-token.spec.ts#L44

We create the component, then detect changes https://github.com/ngxs/store/blob/461a3346e4ec6d2451eb5a7b181270017a0853db/packages/store/tests/state-token.spec.ts#L55

And expect this to happen https://github.com/ngxs/store/blob/461a3346e4ec6d2451eb5a7b181270017a0853db/packages/store/tests/state-token.spec.ts#L61-L63
^ this failed sporadically without `async` for me locally.

Moreover, these are asynchronous operations obviously https://github.com/ngxs/store/blob/461a3346e4ec6d2451eb5a7b181270017a0853db/packages/store/tests/state-token.spec.ts#L68-L78

I would even suggest to rewrite the above like this
```typescript
store
  .select(TODO_LIST_TOKEN)
  .pipe(take(1))
  .subscribe((value: string[]) => {
      expect(value).toEqual(['hello', 'world']);
  });

store
  .selectOnce(TODO_LIST_TOKEN)
  .subscribe((value: string[]) => {
    expect(value).toEqual(['hello', 'world']);
  });
```

Reply:
> As far as I know all specs that interact with streams (if we want to assert something after a stream emits) should be async.

> Moreover, these are asynchronous operations obviously

That would've been true if the stream emits asynchronously. The store emits asynchronously only if there's any asynchronous action being handled, otherwise `store.select` always emits synchronously. This is the same as `store.dispatch` with synchronous actions:
```ts
console.log(1);
store.dispatch(new MySyncAction()).subscribe(() => console.log(2));
console.log(3);

// The logs are 1, 2, 3
```

The `async` (`waitForAsync`) is waiting for all micro and macrotasks to be completed within the `AsyncTestZoneSpec`, and there're no scheduled tasks in this test.

If it's the reason to fail, I would re-write it in a `async-await` manner to be more imperative, I personally don't like having subscriptions and expectations within `subscribe` blocks since you never sure if those expects are actually called (unless we do `expect.hasAssertions()`, but it's not reasonable doing that in all tests).